File: ASP.NET Core + Angular/Task/OnlineBookStoreAPI/Models/Domain/Order.cs
namespace OnlineBookStoreAPI.Models.Domain
{
    public class Order
    {
        public int Id { get; set; }
        public string BillingName { get; set; }
        public string Address { get; set; }
        public string ContactNo { get; set; }
        public string ShippingMethod { get; set; }
        public double TotalPrice { get; set; }
        public DateTime Date { get; set; } = DateTime.Now;
        public string AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        public List<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}

File: ASP.NET Core + Angular/Task/client/src/app/_models/order.ts
import { Photo } from "./photo";

export interface Order {
    id: number;
    billingName: string;
    contactNo: number;
    address: string;
    shippingMethod: string;
    totalPrice: number;
    date: Date;
    orderItems?: OrderItem[];

    // constructor(billingName: string, contactNo: number, address: string, shippingMethod: string, totalPrice: number, id?: number, orderedItems?: OrderItem[]) {
    //     this.id = id || 0;    
    //     this.billingName = billingName;
    //     this.contactNo = contactNo;
    //     this.address = address;
    //     this.shippingMethod = shippingMethod;
    //     this.orderItems = orderedItems || [];
    //     this.totalPrice = totalPrice;
    //     this.date = new Date();
    // }

}

export interface OrderItem {
    quantity: number;   
    totalPrice: number;
    orderBook?: OrderBook; 
}

export interface OrderBook {
    title: string;
    author: string;
    isbn: string;
    photoUrl: string;
    price: number;
    description: string;
    photos: Photo[];

}